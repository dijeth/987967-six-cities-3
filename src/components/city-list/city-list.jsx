import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import ActionCreator from '../../action-creator.js';
import withActiveItem from '../../hocs/with-active-item/with-active-item.jsx';

const CityList = ({items, activeItem, onListClick}) => {
  const cityList = items.map((it, i) => {
    return (
      <li className="locations__item" key={`${it}-${i}`}>
        <a className={`locations__item-link tabs__item ${it === activeItem ? `tabs__item--active` : ``}`} href="#">
          <span data-index={i}>{it}</span>
        </a>
      </li>);
  });

  return <ul className="locations__list tabs__list" onClick={onListClick ? onListClick : null}>{cityList}</ul>;
};

CityList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string).isRequired,
  activeItem: PropTypes.string,
  onActiveItemChange: PropTypes.func,
  onListClick: PropTypes.func
};

const mapDispatchToProps = (dispatch) => ({
  onActiveItemChange(activeCity) {
    dispatch(ActionCreator.changeCity(activeCity));
  }
});

const connectedCityList = connect(null, mapDispatchToProps)(withActiveItem(CityList));

export {CityList};
export default connectedCityList;
