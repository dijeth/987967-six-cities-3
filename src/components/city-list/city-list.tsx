import * as React from 'react';
import {connect} from 'react-redux';
import ActionCreator from '../../reducers/app/action-creator.js';
import withActiveItem from '../../hocs/with-active-item/with-active-item';
import {cityPropType} from '../../const/props.js';
import { City } from '../../interfaces.js';

type Props = {
  items: Array<City>;
  activeItem: City;
  onListClick: (evt: React.MouseEvent<HTMLElement>) => void;
};

const CityList: React.FC<Props> = ({items, activeItem, onListClick}) => {
  if (!items.length) {
    return null;
  }

  const cityName = activeItem !== null ? activeItem.name : null;
  const cityList = items.map((it, i) => {
    return (
      <li className="locations__item" key={`${it}-${i}`}>
        <a className={`locations__item-link tabs__item ${it.name === cityName ? `tabs__item--active` : ``}`} href="#">
          <span data-index={i}>{it.name}</span>
        </a>
      </li>);
  });

  return <ul className="locations__list tabs__list" onClick={onListClick ? onListClick : null}>{cityList}</ul>;
};

const mapDispatchToProps = (dispatch) => ({
  onActiveItemChange(activeCity) {
    dispatch(ActionCreator.changeCity(activeCity.name));
  }
});

const connectedCityList = connect(null, mapDispatchToProps)(withActiveItem<City>(CityList));

export {CityList};
export default connectedCityList;
