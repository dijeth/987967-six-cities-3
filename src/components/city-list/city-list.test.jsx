import React from 'react';
import renderer from 'react-test-renderer';
import {CityList} from './city-list.jsx';

const props = {
  items: [{
    zoom: 1,
    name: `city1`
  }, {
    zoom: 1,
    name: `city2`
  }, {
    zoom: 1,
    name: `city3`
  }, {
    zoom: 1,
    name: `city4`
  }, {
    zoom: 1,
    name: `city5`
  }],
  activeItem: {
    zoom: 1,
    name: `city2`
  },
  onChangeActiveCity: () => {}
};

it(`<CityList /> should be render correctly`, () => {
  const cityList = renderer.create(
      <CityList {...props}/>).toJSON();

  expect(cityList).toMatchSnapshot();
});
