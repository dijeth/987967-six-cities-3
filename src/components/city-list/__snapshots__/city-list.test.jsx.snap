// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityList /> should be render correctly 1`] = `
<ul
  className="locations__list tabs__list"
  onClick={null}
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item "
      href="#"
    >
      <span
        data-index={0}
      >
        city1
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item tabs__item--active"
      href="#"
    >
      <span
        data-index={1}
      >
        city2
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item "
      href="#"
    >
      <span
        data-index={2}
      >
        city3
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item "
      href="#"
    >
      <span
        data-index={3}
      >
        city4
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item "
      href="#"
    >
      <span
        data-index={4}
      >
        city5
      </span>
    </a>
  </li>
</ul>
`;
